sensorprocess/sensorprocess.go

```diff
@@ -109,3 +112,96 @@ func tryAddSensorReading(ctx context.Context, reading []byte, readingTime time.T
 	timeElapsedMs := int(time.Since(startTime).Milliseconds())
 	return int(math.Max(0, float64(config.LidarDataRateMsec-timeElapsedMs)))
 }
+
+// StartIMU polls the IMU to get the next sensor reading and adds it to the cartofacade.
+// stops when the context is Done.
+func StartIMU(
+	ctx context.Context,
+	config Config,
+) bool {
+	for {
+		select {
+		case <-ctx.Done():
+			return false
+		default:
+			if jobDone := addIMUReading(ctx, config); jobDone {
+				return true
+			}
+		}
+	}
+}
+
+// addLidarReading adds a lidar reading to the cartofacade.
+func addIMUReading(
+	ctx context.Context,
+	config Config,
+) bool {
+	tsr, err := config.IMU.TimedIMUSensorReading(ctx)
+	if err != nil {
+		config.Logger.Warn(err)
+		// only end the sensor process if we are in offline mode
+		if config.LidarDataRateMsec == 0 {
+			return strings.Contains(err.Error(), replaypcd.ErrEndOfDataset.Error())
+		}
+		return false
+	}
+
+	sr := cartofacade.IMUReading{
+		LinAccX: tsr.LinearAcceleration.X,
+		LinAccY: tsr.LinearAcceleration.Y,
+		LinAccZ: tsr.LinearAcceleration.Z,
+		AngVelX: tsr.AngularVelocity.X,
+		AngVelY: tsr.AngularVelocity.Y,
+		AngVelZ: tsr.AngularVelocity.Z,
+	}
+	/*
+	 when the lidar data rate msec is 0, we assume the user wants to be in "offline"
```

Comment:
update comment

Reply:
I think my comment here was in regard to using the imu data rate not lidar but that was resolved and comment remains correct